app/models/service_template.rb

```diff
@@ -258,10 +258,35 @@ def self.default_reconfiguration_entry_point
     nil
   end
 
+  def set_resource_actions(config_info, dialog)
+    [
+      {:name      => 'Provision',
+       :param_key => 'fqname',
+       :method    => 'default_provisioning_entry_point',
+       :args      => [service_type]},
+      {:name      => 'Reconfigure',
+       :param_key => 'reconfigure_fqname',
+       :method    => 'default_reconfiguration_entry_point',
+       :args      => []},
+      {:name      => 'Retirement',
+       :param_key => 'retire_fqname',
+       :method    => 'default_retirement_entry_point',
+       :args      => []}
+    ].each do |action|
+      fqname = if config_info[action[:param_key]].nil?
+                 self.class.send(action[:method], *action[:args]) || ""
+               else
+                 config_info[action[:param_key]]
+               end
+      resource_actions.build(:action => action[:name], :fqname => fqname, :dialog => dialog)
+    end
+    save!
+  end
+
```

Comment:
Hi @jntullo, I've looked at this with @gmcculloug a bit,
this method can stay here in app/models/service_template.rb, seems generic enough, the only thing that took a while to understand was the config_info.  

Can you change the signature in this method (and update the calling sequence in API accordingly) as follows:

```
def set_resource_actions(ae_endpoints, dialog)
```

where ae_endpoints would be the following hash:

```
{
    :provisioning => "/.....",
    :retirement => "/.....",
    :reconfigure => "/....."
}
```

non-existent/nil entries honored as you have it today.

Thanks,